resources/building-mobile-extensions/building-an-extension.md

```diff
@@ -23,15 +23,69 @@ Before you build an extension, complete the following tasks:
+## Implementing an Adobe Experience Platform SDK Extension
 
 To create a simple extension, complete the following procedures in the order in which they are listed:
 
 ### A. **Create an Extension Class**
 
-The `ACPExtension`\(iOS\) or `Extension` \(Android\) class is the base class that any extensions must derive from. The `init` method \(iOS\) or the base `constructor` \(Android\) of your extension class is where you will have the opportunity to extend the Adobe Cloud Platform SDKs functionality by registering event listeners, or by setting a default shared state that other modules can access.
+The `ACPExtension`\(iOS\) or `Extension` \(Android\) class is the base class that any extensions must derive from. The `init` method \(iOS\) or the base `constructor` \(Android\) of your extension class is where you will have the opportunity to extend the Adobe Experience Platform SDKs functionality by registering event listeners, or by setting a default shared state that other modules can access.
+
+{% tabs %}
+{% tab title="Android" %}
+
+#### **Android**
+
+The `Extension` class has the following method that you must override:
+
+- `getName`, which returns the name of the extension.
+
+  Extension developers must prefix their extension names with the company name \(for example, _com.myCompany.myExtension_\). For more information about the naming constraints, see Namespace Conventions. The name that you use to register cannot conflict with other registered extensions or Adobe internal modules.
+
+**Tip**: All Adobe module names are prefixed with _com.adobe.module_ and are considered reserved.
+
+- `onUnregistered`, which allows your extension to complete the cleanup that is required when the Adobe Experience Platform SDK unregisters your extension.  Unregistration typically happens at app shutdown but can also occur when an extension is behaving badly. Examples of the extension behaving badly include taking too long to handle a callback or by throwing an exception.
+
+The `Extension` class has the following methods that you can optionally override and a member that provides access to the Event Hub:
+
+- `getVersion`, which returns a version string for your extension. The version string is only used for logging and is currently not validated for formatting.
+
+- `onUnexpectedError`, which allows you log additional information when the Adobe Experience Cloud Platform SDK encounters an error that could not be returned immediately from a call into the Adobe Experience Platform SDK.  An example is an exception that is thrown on a worker thread. The exceptions are rare after your extension has been correctly implemented, but the exceptions might occur during development.
+
+- `getApi` , allows the extension developer to interact with the Event Hub to register event listeners, manage shared state, and so on.
+
+  This method can be used at any after the extension registration is complete. It may also be used by your listeners by using the extension member.
```

Comment:
Keep this paragraph in line with the `getApi` paragraph. With the blank line between it's not clear that this paragraph refers to the `getApi` method.